image-download

```diff
@@ -79,6 +80,20 @@ def check_curl_args(args):
 def find(name, stores, latest, quiet):
     found = []
 
+    try:
+        with open(xdg_config_home('cockpit-dev', 'image-store-preference'), 'r') as fp:
+            pattern = fp.read().strip()
+            matches = [store for store in stores if pattern in store]
+            try:
+                # move it to the front of the list
+                preferred, = matches
```

Comment:
That's not that easy to read, and the comment is odd - here nothing gets "moved", it just picks out the single value out of the list.

Why so strict? Would it hurt to be able to specify "e2e" or "centos" or "linode" and make it select a subset? Then this code here would become simpler, and below you'd just write `if store in preferred` instead of `==`.

Reply:
I had a `remove()` here that disappeared somewhere...

I'll think about what to do.